TruePath/PathStrings.cs

```diff
@@ -36,61 +37,93 @@ public static class PathStrings
     ///     Note that this operation will never perform any file IO, and is purely string manipulation.
     /// </para>
     /// </summary>
+    [SkipLocalsInit] // is necessary to prevent the CLR from filling stackalloc with zeros.
     public static string Normalize(string path)
     {
-        // TODO[#19]: Optimize this. It is possible to do with less allocations.
-        var segments = new List<(int Start, int End)>();
+        int written = 0;
+        Span<char> normalized = stackalloc char[path.Length];
+        ReadOnlySpan<char> source = path.AsSpan();
 
-        int? currentSegmentStart = 0;
-        for (var i = 0; i < path.Length; i++)
+        var buffer = normalized;
+
+        while (true)
         {
-            if (path[i] == Path.DirectorySeparatorChar || path[i] == Path.AltDirectorySeparatorChar)
+            bool last = false;
+            var separator = source.IndexOf(Path.DirectorySeparatorChar);
+            var altSeparator = source.IndexOf(Path.AltDirectorySeparatorChar);
+
+            if (altSeparator == -1 && separator == -1) { last = true; separator = source.Length - 1; }
+            else if (separator == -1) separator = altSeparator;
+            else separator = Math.Min(separator, altSeparator);
```

Comment:
It is missing handling of the case when `altSeparator == -1`:

![image](https://github.com/ForNeVeR/TruePath/assets/55300023/0640fab0-c282-4bef-b2d2-d0d4d6e92aa3)